packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -170,13 +144,59 @@ export const Datepicker: ComponentWithAs<'div', DatepickerProps> &
     todayDateFormatString: props.todayDateFormatString,
   };
 
+  const { calendar, popup, input, className, design, styles, variables, formatMonthDayYear, allowManualInput } = props;
   const valueFormatter = date => (date ? formatMonthDayYear(date, dateFormatting) : '');
 
+  const [openState, setOpenState] = useAutoControlled<boolean>({
+    defaultValue: props.defaultCalendarOpenState,
+    value: props.calendarOpenState,
+    initialValue: false,
+  });
+
+  const [selectedDate, setSelectedDate] = useAutoControlled<Date | undefined>({
+    defaultValue: props.defaultSelectedDate,
+    value: props.selectedDate,
+    initialValue: undefined,
+  });
+  const [formattedDate, setFormattedDate] = React.useState<string>(valueFormatter(selectedDate));
+
+  const restrictedDatesOptions: IRestrictedDatesOptions = {
+    minDate: props.minDate,
+    maxDate: props.maxDate,
+    restrictedDates: props.restrictedDates,
+  };
+
+  const [error, setError] = React.useState<string>(() =>
+    !!props.selectedDate || !!props.defaultSelectedDate
+      ? validateDate(selectedDate, formattedDate, restrictedDatesOptions, dateFormatting, props.required)
+      : '',
+  );
+
+  const calendarOptions: IDayGridOptions = {
+    selectedDate: selectedDate ?? props.today ?? new Date(),
+    navigatedDate: !!selectedDate && !error ? selectedDate : props.today ?? new Date(),
+    firstDayOfWeek: props.firstDayOfWeek,
+    firstWeekOfYear: props.firstWeekOfYear,
+    dateRangeType: props.dateRangeType,
+    daysToSelectInDayView: props.daysToSelectInDayView,
+    today: props.today,
+    showWeekNumbers: props.showWeekNumbers,
+    workWeekDays: props.workWeekDays,
+    ...restrictedDatesOptions,
+  };
+
   const ElementType = getElementType(props);
   const unhandledProps = useUnhandledProps(Datepicker.handledProps, props);
   const getA11yProps = useAccessibility(props.accessibility, {
     debugName: Datepicker.displayName,
-    actionHandlers: {},
+    actionHandlers: {
+      open: e => {
```

Comment:
Looks strange that on click we are alternating between opening and closing it, but with keyboard we are always opening it. Is that really by design? Is it because the focused is moved when opened?

Reply:
Actually, the behavior should be consistent. We keep the calendar open if the manualInput is disabled for both on click and on enter.

If the manualInput is enabled, then we don't handle the enter action in any way.